vite.config.ts

```diff
@@ -35,6 +35,13 @@ export default defineConfig(({ mode }) => {
             '**/test/**',
           ],
           strictOutput: true,
+          beforeWriteFile: (filePath, content) => {
+            // This is a hack to ensure that the i18next.d.ts file is imported in the index.d.ts file, otherwise i18next type augmentation is not available
+            if (filePath.includes('index.d.ts')) {
+              return { content: 'import "./types/i18next.d.ts"\n' + content, filePath }
+            }
```

Comment:
is index.d.ts generated automatically? is that why we need to append to it here vs. editing it directly?